### PR TITLE
Security Fix for Prototype Pollution - huntr.dev

### PR DESCRIPTION
**Pull request by huntr-helper**
https://huntr.dev/users/arjunshibu has fixed the Prototype Pollution vulnerability 🔨. Think you could fix a vulnerability like this?

Get involved at https://huntr.dev/

Q | A
Version Affected | ALL
Bug Fix | YES
Original Pull Request | https://github.com/418sec/libnested/pull/1
Vulnerability README | https://github.com/418sec/huntr/blob/master/bounties/npm/libnested/1/README.md

### :bar_chart: Metadata *

`libnested` is vulnerable to `Prototype Pollution`.

### :gear: Description *

Prototype Pollution refers to the ability to inject properties into existing JavaScript language construct prototypes, such as objects.
JavaScript allows all Object attributes to be altered, including their magical attributes such as `__proto__`, `constructor` and `prototype`. An attacker manipulates these attributes to overwrite, or pollute, a JavaScript application object prototype of the base object by injecting other values. Properties on the Object.prototype are then inherited by all the JavaScript objects through the prototype chain.

### :computer: Technical Description *

Fix implemented by not allowing to modify object prototype.

### :bug: Proof of Concept (PoC) *

1. Create the following PoC file:
```JavaScript
// poc.js
var libnested = require("libnested")
var obj = {}
console.log("Before : " + {}.polluted);
libnested.set(obj, ['__proto__','polluted'], 'Yes! Its Polluted');
console.log("After : " + {}.polluted);
```
2. Execute the following commands in terminal:
```bash
npm i libnested # Install affected module
node poc.js #  Run the PoC
```
3. Check the Output:
```
Before: undefined
After: Yes! Its Polluted
```

### :fire: Proof of Fix (PoF) *

image

### +1 User Acceptance Testing (UAT)

* I've executed unit tests.
* After fix the functionality is unaffected.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ function set (obj, path, value) {
       obj[path[i]] = value
     else if(null == obj[path[i]])
       obj = (obj[path[i]] = isNonNegativeInteger(path[i+1]) ? [] : {})
-    else
+    else if (!(isPrototypePolluted(path[i])))
```
Comment (mixmix) [inline index.js:30]: This looks like a good solution to guarding against prototype mutation

**Review by mixmix — APPROVED** [review]

**Comment by mixmix** [comment]
Have DM'd @ahdinosaur about this, he's going to get to this soon

**Comment by ahdinosaur** [comment]
merged and published as `1.5.1` :tada:

**Comment by mixmix** [comment]
Woop! No more critical npm audit error!

**Pull request merged**